### PR TITLE
AC-555 Added country picker to register patient activity

### PR DESCRIPTION
**Pull request by codepoet2017390**
## Description of what I changed
Added country picker to register patient activity

## Issue I worked on

JIRA Issue: https://issues.openmrs.org/browse/AC-555

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

- [ ] I have **added tests** to cover my changes. (If you refactored
existing code that was well tested you do not have to add tests)

- [x] All new and existing **tests passed**.

- [x] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java [hunk 1]
```diff
@@ -208,6 +208,14 @@ public void scrollToTop() {
         scrollView.smoothScrollTo(0, scrollView.getPaddingTop());
     }
 
+    private boolean isCcpEmpty()
+    {
+        if(ccp.getSelectedCountryName()==null)
```
Comment (sanjulamadurapperuma) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java]: For easier maintainability in the future, I suggest having curly braces for this if condition.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java [hunk 2]
```diff
@@ -208,6 +208,14 @@ public void scrollToTop() {
...
+            return true;
+
+        return false;
+    }
+
```
Comment (sanjulamadurapperuma) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java]: Remove this extra line :)

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java [hunk 3]
```diff
@@ -122,6 +122,7 @@
     private EditText edaddr2;
     private EditText edcity;
     private AutoCompleteTextView edstate;
+    private com.hbb20.CountryCodePicker ccp;
```
Comment (f4ww4z) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java]: @codepoet2017390 Import this instead for future development.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java [hunk 4]
```diff
@@ -208,6 +208,13 @@ public void scrollToTop() {
         scrollView.smoothScrollTo(0, scrollView.getPaddingTop());
     }
 
+    private boolean isCcpEmpty()
+    {
+        if(ccp.getSelectedCountryName()==null) {
```
Comment (f4ww4z) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java]: Please format this file using Android Studio's default formatter.

Comment (f4ww4z) [reply openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java]: It will be better to use a one-liner for this: something like `return ccp.getSelectedCountryName() == null`

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java [hunk 5]
```diff
@@ -434,7 +440,7 @@ private void resolveViews(View v) {
         edaddr2 = (EditText) v.findViewById(R.id.addr2);
         edcity = (EditText) v.findViewById(R.id.city);
         edstate = (AutoCompleteTextView) v.findViewById(R.id.state);
-        edcountry = (AutoCompleteTextView) v.findViewById(R.id.country);
+        ccp= v.findViewById(R.id.ccp);
```
Comment (f4ww4z) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java]: Format this using AS's default formatter. Also always name the variable to something that's self descriptive. e.g. `countryCodePicker`

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java (hunk 3 above)
Comment (codepoet2017390) [reply openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java]: @f4ww4z  are you sure the countries array set up in the app is specially structured according to OpenMRS's policy or It is just a standard set of countries because it contains a standard set of countries which majorly cover all the countries and so does this library.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java [hunk 6]
```diff
@@ -122,42 +122,37 @@
...
     private TextView countryerror;
-
     private Button submitConfirm;
     private Button datePicker;
-
     private DateTimeFormatter dateTimeFormatter;
-
     private String[] countries;
     private ImageView patientImageView;
-
```
Comment (f4ww4z) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java]: @codepoet2017390 Don't remove these lines, they're not part of the PR, and bad formatting.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java [hunk 7]
```diff
@@ -122,42 +122,37 @@
...
     private FloatingActionButton capturePhotoBtn;
     private Bitmap patientPhoto = null;
     private Bitmap resizedPatientPhoto = null;
     private String patientName;
     private File output = null;
-    private final static int IMAGE_REQUEST = 1;
-    private final static int GALLERY_IMAGE_REQUEST = 2;
```
Comment (f4ww4z) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java]: This is not part of the PR.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java [hunk 8]
```diff
@@ -122,42 +122,37 @@
...
     private OpenMRSLogger logger = new OpenMRSLogger();
 
+    public static AddEditPatientFragment newInstance() {
```
Comment (f4ww4z) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java]: What is the functionality of this method? Please remove it and document it in JIRA first.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java [hunk 9]
```diff
@@ -208,6 +203,9 @@ public void scrollToTop() {
         scrollView.smoothScrollTo(0, scrollView.getPaddingTop());
     }
 
+    private boolean isCcpEmpty() {
```
Comment (f4ww4z) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/addeditpatient/AddEditPatientFragment.java]: @codepoet2017390 Move this method to `ViewUtils`, and pass in a country code picker as the parameter, to better organize the code.

**Comment by codepoet2017390** [comment]
@f4ww4z All these non-required changes are due to some problem while squashing commits. I was busy with some University work will update soon.
